### PR TITLE
Dispatch events when invoices are created and deleted through Invoice…

### PR DESCRIPTION
**Pull request by tanishalfelven**
…ServiceBase.php

### CODE REVIEW

**Comment by tanishalfelven** [comment]
Removed data setup, and instead have clean event dispatches for each event call. All dispatches now have scope relevant names.

File: Infusionsoft/InvoiceServiceBase.php [hunk 1]
```diff
@@ -138,10 +140,7 @@ public static function createBlankOrder($contactId, $description, $orderDate, $l
         );
 
         $result = parent::send($app, "InvoiceService.createBlankOrder", $params);
-        $invoice = new Infusionsoft_Invoice($result);
-        $job = new Infusionsoft_Job($invoice->JobId);
-        Infusionsoft_SdkEventManager::dispatch(new Infusionsoft_SdkEvent($invoice, array('result' => $result)), 'DataObject.Saved');
-        Infusionsoft_SdkEventManager::dispatch(new Infusionsoft_SdkEvent($job, array('result' => $result)), 'DataObject.Saved');
+        Infusionsoft_SdkEventManager::dispatch(new Infusionsoft_SdkEvent($result, array('result' => $result)), 'InvoiceService.OrderCreated');
```
Comment (freezey) [inline Infusionsoft/InvoiceServiceBase.php]: There's no reason to send $result twice, the array can be left out and just the invoiceId (which is $result) be sent

File: Infusionsoft/InvoiceServiceBase.php [hunk 2]
```diff
@@ -272,11 +271,8 @@ public static function deleteInvoice($invoiceId, Infusionsoft_App $app = null){
             (int) $invoiceId
         );
 
-        $invoice = new Infusionsoft_Invoice($invoiceId);
-        $job = new Infusionsoft_Job($invoice->JobId);
         $result = parent::send($app, "InvoiceService.deleteInvoice", $params);
-        Infusionsoft_SdkEventManager::dispatch(new Infusionsoft_SdkEvent($invoice, array('result' => $result)), 'DataObject.Deleted');
-        Infusionsoft_SdkEventManager::dispatch(new Infusionsoft_SdkEvent($job, array('result' => $result)), 'DataObject.Deleted');
+        Infusionsoft_SdkEventManager::dispatch(new Infusionsoft_SdkEvent($result, array('result' => $result)), 'InvoiceService.OrderDeleted');
```
Comment (freezey) [inline Infusionsoft/InvoiceServiceBase.php]: $result is going to be a boolean here (I think). Better to send the $invoiceId that was passed in instead.

**Comment by tanishalfelven** [comment]
Those fix your specific comments.... Do you want me to squash it before accepting?

**Pull request merged**